packages/react-dom/src/shared/DOMProperty.js

```diff
@@ -255,25 +278,16 @@ var HTMLDOMPropertyConfig = {
     // See http://schema.org/docs/gs.html
     itemScope: HAS_BOOLEAN_VALUE,
     // These attributes must stay in the white-list because they have
-    // different attribute names (see DOMAttributeNames below)
+    // different attribute names (see `attributeNames`)
+    // TODO: this doesn't seem great? Probably means we depend on
+    // existence in the whitelist somewhere we shouldn't need to.
```

Comment:
I remember when we hit this adding custom attributes, and it didn't feel great. It would be awesome to get to the bottom of this.

Reply:
Yea, definitely will. I have another set of commits that dives into this but it turned out to be a rabbit hole and I decided to do these first.